### ISSUE TITLE
Problematic sawtooth pattern for SOC, RH Dalton Transect Site 1, as CMT 06

### ISSUE THREAD

**Issue by tobeycarman** [ua-snap/dvm-dos-tem#387]
Uncovered by @csilberste when running for Dalton Highway Transect, Site 1 as CMT 06. 

The Dalton Highway Transect Site 1 is located here:
```
h_ds = nc.Dataset("input-staging-area/site_1_1x1/historic-climate.nc")
h_ds.variables['lat'][:]
array([[ 66.92570496]], dtype=float32)

h_ds.variables['lon'][:]
array([[-150.28289795]], dtype=float32)
```

This issue was originally discovered in a run done under PEcAn. I then replicated this run on atlas for testing purposes. I copied the particular run folder from modex to atlas and then adjusted a few paths in the config file. Then did the run like so:
`./dvmdostem -l err --force-cmt 6 -p 100 -e 1000 -s 450 -t 109 -n 91`

When I plot the SOC and RH values, we get this:
![screen shot 2018-08-07 at 3 42 30 pm](https://user-images.githubusercontent.com/838735/43808550-091b9828-9a5a-11e8-9adc-575a35256be7.png)

Here is the code to make the above plot:
```
import netCDF4 as nc
import os
import numpy as np
import matplotlib.pyplot as plt


outs = "input-staging-area/site_1_1x1/output/"

soc_ds = nc.Dataset(os.path.join(outs, "SOC_monthly_tr.nc"))
soc = soc_ds.variables['SOC']
time = soc_ds.variables['time']
  
rh_ds = nc.Dataset(os.path.join(outs, "RH_monthly_tr.nc"))
rh = rh_ds.variables['RH']


fig, ax1 = plt.subplots()

fig.suptitle("SOC and RH for {}".format(outs))

ax1.set_xlabel(time.units)

ax1.plot(soc[:,0,0], label="SOC", color='b')
ax1.set_ylabel("SOC {}".format(soc.units), color='b')
ax1.tick_params('y', colors='b')

ax2 = ax1.twinx()
ax2.plot(rh[:,0,0], label="RH", color='r')
ax2.set_ylabel("RH {}".format(rh.units), color='r')
ax2.tick_params('y', colors='r')
```

**Comment by tobeycarman** [comment]
This is not related to #351 because for the runs that created this sawtooth pattern, the klai parameter is set from PEcAn's meta analysis and ends up within an acceptable range (~0.3)

**Comment by tobeycarman** [comment]
The problem is apparent, but looks a little different in the equlibrium stage:
![screen shot 2018-08-07 at 4 16 04 pm](https://user-images.githubusercontent.com/838735/43809194-4ce26214-9a5d-11e8-8685-ae57f75a3da9.png)

**Comment by serbinsh** [comment]
Question, does the pattern repeat after the same number of X years?  Is it a nested loop index issue perhaps?

Sent from my iPhone

On Aug 7, 2018, at 7:17 PM, tobey <notifications@github.com<mailto:notifications@github.com>> wrote:


The problem is apparent, but looks a little different in the equlibrium stage:
[screen shot 2018-08-07 at 4 16 04 pm]<https://user-images.githubusercontent.com/838735/43809194-4ce26214-9a5d-11e8-8685-ae57f75a3da9.png>

**Comment by EugenieEuskirchen** [comment]
@tobeycarman -I'm not sure how you were considering the best way to tackle this bug, but given the drops in SOC, it might be best to look at that first rather than RH, since based on your last graphs it seems like RH might be related to the drops in SOC. Is that what you are thinking too?

**Comment by tobeycarman** [comment]
@EugenieEuskirchen yeah, that is what I was thinking. I have been on a massive run-around trying to get myself setup with a good debugging setup again. I broke my OSX install and my VM last spring and fall trying to compile for parallel NetCDF, so haven't got a working setup with a graphical debugger. I think I might have gotten that straightened out this morning though...

So my plan has been to stop the debugger right before one of the sawtooth-crashes in SOC and just start looking at the data structures and see if I can find why SOC is plummeting (current idea is that it might be due to layers being created/deleted due to some threshold that is reached). I am surprised to see that RH is spiking after SOC drops though? I was expecting RH to go up preceding or contemporaneous with drop in SOC? I guess this might be indicative of some layer reshuffling though??

**Comment by EugenieEuskirchen** [comment]
@tobeycarman -yes, I remember you mentioning that issue with your OSX and VM. Hope you got it squared away without too much of a headache.
I think that's a good plan in terms of the debugging. I also thought it was strange it took a month for the RH to peak, as though there is some sort of lag that shouldn't be there. So, yes, could be a layer reshuffling issue.

**Comment by tobeycarman** [comment]
Here are some more plots, generated with some plotting code that has not yet been comitted.

These plots use X axis for time. Y axis is for the value being plotted (i.e. SOC, or RH or similar). The color of the layer indicates its index. So lighter, yellower for layers closer to the surface, darker purple for deeper layers. In plots with multiple panels, the top panel is is a stacked area chart showing the sum of all layers. The other panels show a filled area for the individual layers.

**Comment by tobeycarman** [comment]
This shows all layers for SOC and RH. The y scales are shared amongst the panels in each plot so you an see the relative magnitude/contribution from each layer.
<img width="1440" alt="screen shot 2018-09-05 at 9 09 16 am" src="https://user-images.githubusercontent.com/838735/45118034-abca9a00-b103-11e8-95d7-e74fc101d11a.png">

**Comment by tobeycarman** [comment]
Same as above but without sharing the y scales, so you can see the detail better for the layers:
<img width="1440" alt="screen shot 2018-09-05 at 12 05 06 pm" src="https://user-images.githubusercontent.com/838735/45118144-0b28aa00-b104-11e8-8542-ca89cf9c302c.png">

**Comment by tobeycarman** [comment]
And here is a view of just the stacked area chart w/o the individual layer details. This is verification that the drop in SOC is occurring before RH increases.
<img width="1399" alt="screen shot 2018-09-05 at 10 13 46 am" src="https://user-images.githubusercontent.com/838735/45118198-49be6480-b104-11e8-880a-ab31120665e4.png">

**Comment by tobeycarman** [comment]
Check that SOC drops before RH in sites and community types beside Dalton Highway Site 1 CMT 6: yes, SOC is still decreasing before the annual spike in RH.

For this run I used the defaults for the SouthBarrow sample dataset. So pixel 9,9 which is CMT 6.
<img width="1438" alt="screen shot 2018-09-11 at 3 43 25 pm" src="https://user-images.githubusercontent.com/838735/45393671-7f1df300-b5d9-11e8-8860-2ee6cf0418d1.png">


Other observations:

* For SouthBarrow RH occurs higher in the soil column.
* For SouthBarrow, SOC is distributed relatively evenly thru the soil column.
* SOC in South Barrow, cmt6  is an order of magnitude higher than the Dalton Hwy Site 1, cmt 6
* RH in South Barrow cmt 6 is lower and closer to the surface than the Dalton Hwy site cmt 6.

**Comment by tobeycarman** [comment]
Oh - one thought: maybe I am reading these plots incorrectly. It looks like there is a drop in SOC, but maybe what is actually going on is that there is are 2 new layers added at the top of the stack during the summer months.

**Comment by EugenieEuskirchen** [comment]
@tobeycarman -okay, great point - the adding of layers might make some sense, particularly if we consider the litter layer getting deeper at the end of the growing season.  I wonder if we simply looked at the annual values if things would make more sense. We usually don't look at SOC at a monthly level, and ditto for RH.
